contracts-periphery/src/test/UmbraBatchSend.t.sol

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "./utils/DSTestPlus.sol";
+import "../UmbraBatchSend.sol";
+
+interface UmbraToll {
+  function toll() external returns(uint256);
+}
+
+contract UmbraBatchSendTest is DSTestPlus {
+  using stdStorage for StdStorage;
+  StdStorage stdstore;
+
+  address umbra;
+  UmbraBatchSend router;
+  MockERC20 token;
+
+  address tokenAddr;
+  address alice = address(0x202204);
+  address bob = address(0x202205);
+
+  uint256 alicePrevBal;
+  uint256 bobPrevBal;
+  uint256 umbraPrevBal;
+
+  uint256 toll;
+  bytes32 pkx = "pkx";
+  bytes32 ciphertext = "ciphertext";   
+
+  UmbraBatchSend.SendEth[] sendEth;
+  UmbraBatchSend.SendToken[] sendToken;
+
+  error ValueMismatch();
+
+  function setUp() virtual public {
+    umbra = deployCode("src/test/utils/Umbra.json", bytes(abi.encode(0, address(this), address(this))));
+    vm.etch(0xFb2dc580Eed955B528407b4d36FfaFe3da685401, umbra.code);
+    umbra = 0xFb2dc580Eed955B528407b4d36FfaFe3da685401;
+    router = new UmbraBatchSend(address(umbra));
+    token = new MockERC20("Test","TT", 18);
+    tokenAddr = address(token);
+    toll = UmbraToll(umbra).toll();
```

Comment:
Let's delete this line and only explicitly set `toll` in the child contracts so the intent is more clear